% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/ematAdjust.R
\name{ematAdjust}
\alias{ematAdjust}
\title{matrix row-wise scaling and centering}
\usage{
ematAdjust(
  emat,
  center = TRUE,
  scale = TRUE,
  signalFilt = 0,
  verbose = getOption("verbose"),
  ...
)
}
\arguments{
\item{emat}{a numeric matrix with row features and sample columns.}

\item{center}{numeric of same length as \code{nrow(emat)}.}

\item{scale}{numeric of same length as \code{nrow(emat)}.}

\item{signalFilt}{numeric setting feature filtering. Specifically rows
where \code{rowMax(emat) < quantile(emat, probs = signalFilt)} are discarded.}

\item{verbose}{logical, console messages output.}

\item{...}{additional arguments passed to normalizeBetweenArrays and
calcNormFactors depending normalization method selected.}
}
\value{
A row-wise centered and scaled matrix. Output matrix will have fewer rows
than input if signalFilt>0.
}
\description{
Centers and scales gene expression matrix so that each row
has mean=0 and sd=1. If input data is non-normalized sequencing \emph{count}
data, normalization should be performed by setting \code{normMethod}. This
also enforces log2-transformation.
}
\details{
\code{ematAdjust} performs row-wise scaling and centering by passing matrix
to \code{\link[base]{scale}}. Setting \code{scale} and \code{center} may be
useful for predicting new samples based on row-wise means and standard
deviations from prior (identically processed) datasets. If \emph{e.g.}
\code{signalFilt=.1} features with maximum below emat 10th
percentile are discarded.
}
